logging/runtime/src/main/java/io/quarkiverse/googlecloudservices/logging/runtime/util/StackTraceArrayRenderer.java

```diff
@@ -0,0 +1,123 @@
+package io.quarkiverse.googlecloudservices.logging.runtime.util;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.IdentityHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import com.google.common.base.Strings;
+
+import io.quarkiverse.googlecloudservices.logging.runtime.LoggingConfiguration.StackElementRendering;
+
+/**
+ * This is a utility class for rendering a stack trace as an array
+ * of thrown exceptions. It uses the same calculation as {@link Throwable}
+ * for truncating stacks for causes - i.e. not show the same stack frames when
+ * they are common - and checks for self referencing and circular references.
+ */
+public class StackTraceArrayRenderer {
```

Comment:
Is there really a use case for adding the stack trace as an array of frames instead a single string ?
If it's a rare use case, I prefere to avoid addind it.

Reply:
As I said, I got a bit carried away :-D I'll stash it and we can revive it later if someone asks for it.